apps/examples/araplayer/MediaBufferInputDataSource.cpp

```diff
@@ -0,0 +1,115 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <media/MediaTypes.h>
+#include "MediaBufferInputDataSource.h"
+#include "AudioStream.h"
+#include "AudioPDStream.h"
+#include "AudioHASStream.h"
+#include "AudioUtils.h"
+
+MediaBufferInputDataSource::MediaBufferInputDataSource() :
+	isOpened(false)
+{
+	memset(mBuf, 0x00, 4096);
+}
+
+MediaBufferInputDataSource::MediaBufferInputDataSource(audio_type_t audioType, audio_format_type_t format, unsigned int rate, unsigned int chan, void *stream)
+{
+	setChannels(chan);
+	setSampleRate(rate);
+	setPcmFormat(format);
+	setAudioType(audioType);
+	mStream = stream;
+}
+
+MediaBufferInputDataSource::~MediaBufferInputDataSource()
+{
+	if (isPrepared()) {
+		close();
+	}
+}
+
+bool MediaBufferInputDataSource::isPrepared()
+{
+	return isOpened;
+}
+
+bool MediaBufferInputDataSource::open()
+{
+	isOpened = true;
```

Comment:
close checked open state but open doesn't.
Is it correct?

Reply:
yes, There is nothing to check when opening.
but, need a return value on an open iteration.